packages/backend/src/api/jobScraping/scraping.ts

```diff
@@ -0,0 +1,353 @@
+import * as cheerio from 'cheerio';
+import puppeteer from 'puppeteer';
+
+const xpert4 = 'https://xperts4.com/job/?filter-category=30&filter-date-posted=14days';
+const remocate =
+  'https://www.remocate.app/?Search=developer&Location=%F0%9F%93%8D+Any+Location&Category=%F0%9F%8C%9F+Any+Category#job-board';
+const expertiseRecruitment = 'https://jobs.expertiserecruitment.com/jobs/Careers';
+const smartRecruiters = 'https://jobs.smartrecruiters.com/?keyword=Developer&locationType=remote';
+const bayt =
+  'https://www.bayt.com/en/lebanon/jobs/developer-jobs/?filters%5Bjb_last_modification_date_interval%5D%5B%5D=3';
+const remoteSource = 'https://jobs.remotesource.com/jobs?jobTypes=Software+Engineer&postedSince=P1D';
+
+function filterLanguagesAndFrameworks(paragraph: string) {
+  const languages = [
+    'JavaScript',
+    'TypeScript',
+    'Objective-C',
+    'C++',
+    'C#',
+    'Python',
+    'Java',
+    'Ruby',
+    'PHP',
+    'Swift',
+    'Go',
+    'Rust',
+    'Kotlin',
+    'Dart',
+    'R',
+    'C',
+    'Perl',
+    'Scala',
+    'Haskell',
+  ];
+
+  const frameworks = [
+    'React',
+    'Vue.js',
+    'Angular',
+    'Svelte',
+    'Next.js',
+    'Express',
+    'Django',
+    'Flask',
+    'Spring Boot',
+    'Ruby on Rails',
+    'Laravel',
+    'Symfony',
+    'ASP.NET',
+    'Nuxt.js',
+    'Nest.js',
+    'Flutter',
+    'React Native',
+    'Electron',
+    'Meteor',
+    'Backbone.js',
+    'Ember.js',
+    'Fast API',
+    'Bootstrap',
+    '.NET Core',
+  ];
+
+  // Ensure special characters are escaped (C++, C#)
+  const escapedLanguages = languages.map((lang) => lang.replace(/[.*+?^${}()|[\]\\]/g, '\\$&'));
+  const escapedFrameworks = frameworks.map((framework) => framework.replace(/[.*+?^${}()|[\]\\]/g, '\\$&'));
+
+  // Join into a regex that matches any of the languages or frameworks
+  const languagesRegex = new RegExp(`(?<![\\w])(${escapedLanguages.join('|')})(?![\\w])`, 'gi');
+  const frameworksRegex = new RegExp(`(?<![\\w])(${escapedFrameworks.join('|')})(?![\\w])`, 'gi');
+
+  const foundLanguages = paragraph.match(languagesRegex) || [];
+  const foundFrameworks = paragraph.match(frameworksRegex) || [];
+
+  return {
+    languages: [...new Set(foundLanguages)], // Removing duplicates
+    frameworks: [...new Set(foundFrameworks)], // Removing duplicates
+  };
+}
+
+async function shortenLink(link: any) {
+  const shortenedLink = await tinyURLShorten(link);
+  return shortenedLink;
+}
+
+async function tinyURLShorten(link: any) {
+  const response = await fetch(`https://tinyurl.com/api-create.php?url=${encodeURIComponent(link)}`);
+  const shortenedLink = await response.text();
+  return shortenedLink;
+}
+
+export const xpert4Scrape = async () => {
```

Comment:
for each Concrete website we are scraping, let's create a concrete class that implements an interface `Scraper` that implements for now 1 method `execute` 

within execute, I expect to have the scraping logic and implementation